### DISCUSSION TITLE
Should warn if sending bad actions to `pure`, like `undefined`, `false`, etc

### DISCUSSION THREAD

**Discussion by whitelizard** [statelyai/xstate#3399] [General]
XState should give a warning (or something) if `pure` receives values of the wrong data type.

I got really stuck when I had a guard in my mapping of an array into `send`s for `pure`, like this:

```js
pure(({ items }) => items.map(({ ref }) => ref && send('GO', { to: ref })))
```

**Comment by davidkpiano** [comment]
Are you using TypeScript?

**Reply by whitelizard** [reply]
No
